presto-spi/src/main/java/com/facebook/presto/spi/connector/Connector.java

```diff
@@ -160,8 +160,9 @@ default ConnectorAccessControl getAccessControl()
      * Commit the transaction. Will be called at most once and will not be called if
      * {@link #rollback(ConnectorTransactionHandle)} is called.
      */
-    default void commit(ConnectorTransactionHandle transactionHandle)
+    default ConnectorCommitHandle commit(ConnectorTransactionHandle transactionHandle)
     {
+        return null;
```

Comment:
Why null? Some usages rely on this default function, like in some tests; therefore, I cannot throw the UnsupportedOperationException() here. Alternatives include: (1) Optional , (2) ConnectorCommitHandle is a base class instead of an interface; then we can return any empty object. What do you think? @highker

Reply:
- You don't have to return null as long as you remove the `default` tag of the function.
- You will need to change all other connectors to implement this interface.
- Can we also change the callsite of this function to return `ConnectorCommitHandle`?